scripts/main/lychee.js

```diff
@@ -875,38 +1120,68 @@ lychee.setMode = function (mode) {
 			.unbind(["command+backspace", "ctrl+backspace"])
 			.unbind(["command+a", "ctrl+a"]);
 	}
-	if (!lychee.rights.is_admin) {
-		$("#button_users, #button_logs, #button_diagnostics").remove();
+	if (!lychee.rights.is_admin || mode === "view" || mode === "frame") {
+		$("#button_users, #button_logs, #button_diagnostics").hide();
 	}
 
+	const bodyClasses = document.querySelector("body").classList;
+
 	if (mode === "logged_in") {
+		if (!bodyClasses.contains("mode-gallery")) {
+			bodyClasses.replace("mode-none", "mode-gallery");
+			bodyClasses.replace("mode-frame", "mode-gallery");
+			bodyClasses.replace("mode-view", "mode-gallery");
+		}
 		// After login the keyboard short-cuts to login by password (l) and
 		// by key (k) are not required anymore, so we unbind them.
 		Mousetrap.unbind(["l"]).unbind(["k"]);
 
 		// The code searches by class, so remove the other instance.
-		$(".header__search, .header__clear", ".header__toolbar--public").remove();
+		$(".header__search, .header__clear", ".header__toolbar--public").hide();
 
 		if (!lychee.editor_enabled) {
-			$("#button_rotate_cwise").remove();
-			$("#button_rotate_ccwise").remove();
+			$("#button_rotate_cwise").hide();
+			$("#button_rotate_ccwise").hide();
 		}
 		return;
 	} else {
-		$(".header__search, .header__clear", ".header__toolbar--albums").remove();
-		$("#button_rotate_cwise").remove();
-		$("#button_rotate_ccwise").remove();
+		$(".header__search, .header__clear", ".header__toolbar--albums").hide();
+		$("#button_rotate_cwise").hide();
+		$("#button_rotate_ccwise").hide();
 	}
```

Comment:
```suggestion
	$(".header__search, .header__clear", ".header__toolbar--albums").hide();
	$("#button_rotate_cwise").hide();
	$("#button_rotate_ccwise").hide();
```

Considering that the body following the `if` condition above unconditionally returns, the `else` here is very confusing, especially since it doesn't extend through the rest of the method (yes, I know that the code had already been structured that way before).